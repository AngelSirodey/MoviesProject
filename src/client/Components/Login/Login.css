.Login {
    display: flex;
    padding: 30px;
    border: solid 2px #1cbacc;
    border-radius: 6px;
    align-items: center;
    width: fit-content;
    position: absolute;
    right: 0;
    background-color: #ffffff;
    z-index: 1;
}

.Login__img {
    flex: 2;
    width: auto;
}

.form {
    flex: 2;
    display: flex;
    flex-direction: column;
    margin: 0px 32px;
}

.Login__container {
    display: flex;
    flex-direction: column;
    align-items: center;
    padding-bottom: 18px;
}

.Login__container > label {
    font-size: 20px;
    padding-bottom: 6px;
}

.Login__button {
    font-size: 16px;
    background-color: #1cbacc;
    border-radius: 4px;
    padding: 6px 32px;
    border: none;
    color: #ffffff;
    align-self: center;
    margin-top: 12px;
}

.Login__button:hover {
    background-color: rgba(0,0,0,.25);
    cursor: pointer;
  }


@media screen and (max-width: 449px) {
    .Login {
        flex-direction: column;
        padding: 0;
    }
    .form {
      margin: 12px 32px;
    }
}
