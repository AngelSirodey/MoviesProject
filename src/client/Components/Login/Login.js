import React from 'react';

import './Login.css';

const Login = ({ setIsAuth, setOpenLogin }) => {
  const validateLogin = (email, password) => {
    const emailMatch = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
    if  ( emailMatch.test(email) && password !== '' ) {
      return true;
    };
    return false;
  };

  const handleLogin = (event) => {
      event.preventDefault();
      let email = event.target.elements.email.value;
      let password = event.target.elements.password.value;
      const validateUser =  validateLogin(email, password);
      setIsAuth(validateUser);
      setOpenLogin(false);
  };

  return ( 
    <div className="Login">
        <img src="movies.png" className="Login__img" alt="login" />
        <form onSubmit={handleLogin} className="form">
            <div className="Login__container">
              <label htmlFor="email"> Email </label>
              <input type="text" id="email" name="email" />
            </div>
            <div className="Login__container">
              <label htmlFor="password"> Password </label>
              <input type="text" id="password" name="password" />
            </div>
            <button type="submit" className="Login__button">
              Login
            </button>
        </form>
    </div>
  );
};

export default Login;
