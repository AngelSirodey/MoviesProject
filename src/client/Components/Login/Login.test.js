import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Login from './Login';

Enzyme.configure({ adapter: new Adapter() });

let wrapper;

describe('Login behaviour', () => {
    it('should render Login components', () => {
        wrapper = shallow(<Login />);
        expect(wrapper.find('.Login')).toBeTruthy(); ;
    });

    it('should call handleLogin func on submit', () => {
        wrapper = shallow(<Login />);
        expect(wrapper.find('button')).toHaveLength(1);
        wrapper.find('.Login__button').simulate('click');
    });

    it('should render as expected', () => {
        expect(wrapper).toMatchSnapshot();
    });
});
