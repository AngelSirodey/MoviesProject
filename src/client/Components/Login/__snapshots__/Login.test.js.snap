// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login behaviour should render as expected 1`] = `
<div
  className="Login"
>
  <img
    alt="login"
    className="Login__img"
    src="movies.png"
  />
  <form
    className="form"
    onSubmit={[Function]}
  >
    <div
      className="Login__container"
    >
      <label
        htmlFor="email"
      >
         Email 
      </label>
      <input
        id="email"
        name="email"
        type="text"
      />
    </div>
    <div
      className="Login__container"
    >
      <label
        htmlFor="password"
      >
         Password 
      </label>
      <input
        id="password"
        name="password"
        type="text"
      />
    </div>
    <button
      className="Login__button"
      type="submit"
    >
      Login
    </button>
  </form>
</div>
`;
